utils/archive_explorer/tests/test_archive_explorer.py

```diff
@@ -0,0 +1,151 @@
+"""
+Test the archive explorer
+"""
+import datetime
+import os
+import shutil
+import time
+import unittest
+
+from utils.data_archive_creator.data_archive_creator import DataArchiveCreator
+from utils.archive_explorer.archive_explorer import ArchiveExplorer
+
+
+# pylint:disable=invalid-name
+class TestArchiveExplorer(unittest.TestCase):
+    """
+    Test all the functionality of the archive explorer
+    """
+
+    test_output_directory = None
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        Create the test output directory and the data archive directory
+        """
+        cls.test_output_directory = os.path.join(os.path.dirname(os.path.realpath(__file__)),
```

Comment:
Could we instead use `tempfile.mkdtmp` for system tests [see](https://docs.python.org/2/library/tempfile.html#tempfile.mkdtemp). This avoids the potential of running into race conditions down the road

Reply:
Yeah this is a much nicer way to do this. I'll use this in future - thanks for the tip 😄